_studio/shared/umc/codec/h265_dec/src/umc_h265_slice_decoding.cpp

```diff
@@ -317,6 +386,122 @@ int H265Slice::getNumRpsCurrTempList() const
   return numRpsCurrTempList;
 }
 
+void H265Slice::setRefPOCListSliceHeader()
+{
+    int32_t  RefPicPOCSetStCurr0[MAX_NUM_REF_PICS];
+    int32_t  RefPicPOCSetStCurr1[MAX_NUM_REF_PICS];
+    int32_t  RefPicPOCSetLtCurr[MAX_NUM_REF_PICS];
+    uint32_t NumPicStCurr0 = 0;
+    uint32_t NumPicStCurr1 = 0;
+    uint32_t NumPicLtCurr = 0;
+    uint32_t  i;
+
+    for (i = 0; i < getRPS()->getNumberOfNegativePictures(); i++)
+    {
+        if (getRPS()->getUsed(i))
+        {
+            RefPicPOCSetStCurr0[NumPicStCurr0] = m_SliceHeader.m_poc + getRPS()->getDeltaPOC(i);
+            NumPicStCurr0++;
+        }
+    }
+    m_SliceHeader.m_numPicStCurr0 = NumPicStCurr0;
+
+    for (; i < getRPS()->getNumberOfNegativePictures() + getRPS()->getNumberOfPositivePictures(); i++)
+    {
+        if (getRPS()->getUsed(i))
+        {
+            RefPicPOCSetStCurr1[NumPicStCurr1] = m_SliceHeader.m_poc + getRPS()->getDeltaPOC(i);
+            NumPicStCurr1++;
+        }
+    }
+    m_SliceHeader.m_numPicStCurr1 = NumPicStCurr1;
+
+    for (i = getRPS()->getNumberOfNegativePictures() + getRPS()->getNumberOfPositivePictures();
+        i < getRPS()->getNumberOfNegativePictures() + getRPS()->getNumberOfPositivePictures() + getRPS()->getNumberOfLongtermPictures(); i++)
+    {
+        if (getRPS()->getUsed(i))
+        {
+            RefPicPOCSetLtCurr[NumPicLtCurr] = getRPS()->getPOC(i);
+            NumPicLtCurr++;
+        }
+    }
+    m_SliceHeader.m_numPicLtCurr = NumPicLtCurr;
+
+    H265PicParamSet const* pps = GetPicParam();
+    VM_ASSERT(pps);
+
+    int32_t numPicTotalCurr = NumPicStCurr0 + NumPicStCurr1 + NumPicLtCurr;
+    if (pps->pps_curr_pic_ref_enabled_flag)
+        numPicTotalCurr++;
+
+    int32_t rpsPOCCurrList1[MAX_NUM_REF_PICS + 1];
```

Comment:
if you use in-the-middle declarations of variables, do that closer to real usage. This particular one needs to be moved below closer to lie 455 where you first start using it.

Reply:
And I believe you better to initialize it with 0es to satisfy static analyzers in advance...